### ISSUE TITLE
Improve `DatabaseCleaner.cleaners` vs `DatabaseCleaners.connections` deprecation

### ISSUE THREAD

**Issue by deivid-rodriguez** [DatabaseCleaner/database_cleaner#655]
When using `cucumber-rails`, I get a deprecation that reads

```
Calling `DatabaseCleaner.connections` is deprecated, and will be removed in database_cleaner 2.0. Use `DatabaseCleaner.cleaners`, instead.
```

This deprecation makes you think that `DatabaseCleaners.cleaners` is a drop-in replacement for `DatabaseCleaners.connections`, but replacing it makes things crash.

I believe `DatabaseCleaner.cleaners.values` would be a more appropriate recommendation?

**Comment by botandrose** [comment]
@deivid-rodriguez Hmm, good point. If you submit a PR for this, I'll merge it!

Also, just out of curiosity, what is your use-case for calling the `DatabaseCleaners.connections` method? I didn't honestly expect anyone to actually see this deprecation!

**Comment by deivid-rodriguez** [comment]
No idea, the warning comes from `cucumber-rails` code, I was on a round of fixing deprecations in my test suite and decided to try fix this one. But I'm not sure why there're using it, they admittedly call it a hack :laughing:.

This is my PR to `cucumber-rails` where you can see the code: https://github.com/cucumber/cucumber-rails/pull/463.

You might have a better suggestion :)

**Comment by deivid-rodriguez** [comment]
Which branch should I target the PR to? Do you have a `1-8-stable` branch or something like that? I'm asking because the method has been completely removed from master.

**Comment by deivid-rodriguez** [comment]
Oh, I just saw those, there's `v1-8-maintenance` and `v1-maintenance`. Which one?

**Comment by botandrose** [comment]
@deivid-rodriguez Ah, very interesting. They are trying to save the original strategy, so that they can temporarily change it, and then reset it back to the original afterwards. As the existing `# that feels like a nasty hack` comment indicates, this only works when there is only one strategy in play. If there are multiple databases with different strategies, this will stomp on all of them! I think this is an indication of a missing API in DatabaseCleaner. Something like `.cleaning_with(strategy) { ... }`. Something to think about for v2.1, probably! </ramble>

**Comment by botandrose** [comment]
Closed by #656
